src/lib/SX1280Driver/SX1280.cpp

```diff
@@ -0,0 +1,344 @@
+        break;
+
+    case SX1280_MODE_CAD:
+        break;
+
+    default:
+        break;
+    }
+
+    currOpmode = OPmode;
+}
+
+void SX1280Driver::ConfigModParams(SX1280_RadioLoRaBandwidths_t bw, SX1280_RadioLoRaSpreadingFactors_t sf, SX1280_RadioLoRaCodingRates_t cr)
+{
+    // Care must therefore be taken to ensure that modulation parameters are set using the command
+    // SetModulationParam() only after defining the packet type SetPacketType() to be used
+
+    WORD_ALIGNED_ATTR uint8_t rfparams[3] = {0}; //TODO make word alignmed
+
+    rfparams[0] = (uint8_t)sf;
+    rfparams[1] = (uint8_t)bw;
+    rfparams[2] = (uint8_t)cr;
+
+    hal.WriteCommand(SX1280_RADIO_SET_MODULATIONPARAMS, rfparams, sizeof(rfparams));
+}
+
+void SX1280Driver::SetFrequency(uint32_t Reqfreq)
+{
+    //Serial.println(Reqfreq);
+    WORD_ALIGNED_ATTR uint8_t buf[3] = {0}; //TODO make word alignmed
+
+    uint32_t freq = (uint32_t)((double)Reqfreq / (double)SX1280_FREQ_STEP);
+    buf[0] = (uint8_t)((freq >> 16) & 0xFF);
+    buf[1] = (uint8_t)((freq >> 8) & 0xFF);
+    buf[2] = (uint8_t)(freq & 0xFF);
+
+    hal.WriteCommand(SX1280_RADIO_SET_RFFREQUENCY, buf, sizeof(3));
+    currFreq = Reqfreq;
+}
+
+int32_t SX1280Driver::GetFrequencyError()
+{
+    WORD_ALIGNED_ATTR uint8_t efeRaw[3] = {0}; //TODO make word alignmed
+    uint32_t efe = 0;
+    double efeHz = 0.0;
+
+    efeRaw[0] = hal.ReadRegister(SX1280_REG_LR_ESTIMATED_FREQUENCY_ERROR_MSB);
+    efeRaw[1] = hal.ReadRegister(SX1280_REG_LR_ESTIMATED_FREQUENCY_ERROR_MSB + 1);
+    efeRaw[2] = hal.ReadRegister(SX1280_REG_LR_ESTIMATED_FREQUENCY_ERROR_MSB + 2);
+    efe = (efeRaw[0] << 16) | (efeRaw[1] << 8) | efeRaw[2];
+
+    efe &= SX1280_REG_LR_ESTIMATED_FREQUENCY_ERROR_MASK;
+
+    //efeHz = 1.55 * (double)complement2(efe, 20) / (1600.0 / (double)GetLoRaBandwidth() * 1000.0);
+    return efeHz;
+}
+
+void SX1280Driver::SetFIFOaddr(uint8_t txBaseAddr, uint8_t rxBaseAddr)
+{
+    uint8_t buf[2];
+
+    buf[0] = txBaseAddr;
+    buf[1] = rxBaseAddr;
+    hal.WriteCommand(SX1280_RADIO_SET_BUFFERBASEADDRESS, buf, 2);
+}
+
+void SX1280Driver::SetDioIrqParams(uint16_t irqMask, uint16_t dio1Mask, uint16_t dio2Mask, uint16_t dio3Mask)
+{
+    uint8_t buf[8];
+
+    buf[0] = (uint8_t)((irqMask >> 8) & 0x00FF);
+    buf[1] = (uint8_t)(irqMask & 0x00FF);
+    buf[2] = (uint8_t)((dio1Mask >> 8) & 0x00FF);
+    buf[3] = (uint8_t)(dio1Mask & 0x00FF);
+    buf[4] = (uint8_t)((dio2Mask >> 8) & 0x00FF);
+    buf[5] = (uint8_t)(dio2Mask & 0x00FF);
+    buf[6] = (uint8_t)((dio3Mask >> 8) & 0x00FF);
+    buf[7] = (uint8_t)(dio3Mask & 0x00FF);
+
+    hal.WriteCommand(SX1280_RADIO_SET_DIOIRQPARAMS, buf, 8);
+}
+
+void SX1280Driver::ClearIrqStatus(uint16_t irqMask)
+{
+    uint8_t buf[2];
+
+    buf[0] = (uint8_t)(((uint16_t)irqMask >> 8) & 0x00FF);
+    buf[1] = (uint8_t)((uint16_t)irqMask & 0x00FF);
+
+    hal.WriteCommand(SX1280_RADIO_CLR_IRQSTATUS, buf, 2);
+}
+
+void SX1280Driver::TXnbISR()
+{
+    //endTX = micros();
```

Comment:
Can commented out code go?

Reply:
I want to wrap it in an ifdef block so I can debug the onAirTime gonna leave it for now